modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/js/ckeditor5/plugins/Timestamp.tsx

```diff
@@ -0,0 +1,35 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ButtonView, Plugin} from 'ckeditor5';
+
+class Timestamp extends Plugin {
+	init() {
+		const editor = this.editor;
+
+		editor.ui.componentFactory.add('timestamp', () => {
+			const button = new ButtonView();
+
+			button.set({
+				label: Liferay.Language.get('timestamp'),
+				withText: true,
+			});
+
+			button.on('execute', () => {
+				const now = new Date();
+
+				editor.model.change((writer) => {
+					editor.model.insertContent(
+						writer.createText(now.toString())
+					);
+				});
+			});
+
+			return button;
+		});
```

Comment:
Plugins API is a custom JS API, very different from CKE4. Existing custom plugins will need to be rewritten from scratch.